### PR TITLE
TMDM-13179 [Postgres] It's not possible to activate SSL communication between MDM Server and Database

### PR DESCRIPTION
**Pull request by hw-zhu**
https://jira.talendforge.org/browse/TMDM-13179
**What is the current behavior?** (You should also link to an open issue here)
Cannot support any parameter in the connection url, If the URL already contains a parameter then new  parameter should not be append by a '?' but by a `&`.

**What is the new behavior?**
Using new method to create Connection object, that the new Properties argument can be used to pass arbitrary string tag/value pairs as connection arguments.Normally at least "user" and "password" properties should be included in the Properties object. 

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?

**What kind of change does this PR introduce?**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java [hunk 1]
```diff
@@ -45,7 +45,11 @@ public void clean(Storage storage) {
             }
 
             Driver driver = (Driver) Class.forName(dataSource.getDriverClassName()).newInstance();
-            Connection connection = driver.connect(dataSource.getInitConnectionURL() + "?user=" + dataSource.getInitUserName() + "&password=" + dataSource.getInitPassword(), new Properties());  //$NON-NLS-1$ //$NON-NLS-2$
+            Properties properties = new Properties();
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java]: Just not sure if this kind of code section can be retrieved in a Util class?

Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java]: How to refine the code snippet to Util? do you have any idea?

Comment (VictorLi-talendbj) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/prepare/PostgresStorageCleaner.java]:
```
            Driver driver = (Driver) Class.forName(dataSource.getDriverClassName()).newInstance();
            Properties properties = new Properties();
            properties.put("user", dataSource.getInitUserName()); //$NON-NLS-1$
            properties.put("password", dataSource.getInitPassword()); //$NON-NLS-1$

            Connection connection = driver.connect(dataSource.getInitConnectionURL(), properties);
```
Can this part be added in a Util function, such as ```Connection getDBConnection(DataSource dataSource)```?

File: org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java [hunk 2]
```diff
@@ -412,6 +415,65 @@ public Properties getAdvancedPropertiesIncludeUserInfo() {
...
+        Driver driver = createDriver(dataSource);
+        Properties properties = new Properties();
+        properties.put("user", dataSource.getUserName()); //$NON-NLS-1$
+        properties.put("password", dataSource.getPassword()); //$NON-NLS-1$
+
+        String connectionURL = dataSource.getConnectionURL();
+        return driver.connect(connectionURL, properties);
+    }
+
+    private static String getConnectionURL(RDBMSDataSource dataSource) {
```
Comment (VictorLi-talendbj) [inline org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: Can we also name this function to `getInitConnectionURL`?

Comment (hw-zhu) [reply org.talend.mdm.core.storage.sql/src/main/java/com/amalto/core/storage/datasource/RDBMSDataSource.java]: done

**Pull request merged**